department-of-veterans-affairs/vets-website
# MHV Secondary Nav can have links outside an app

## Summary
This change allows the MHV secondary nav to have item links that point to a location outside an app's root URL. This is done by the addition of a new property called `appRootUrl` that can be different from the `href` property. For example, we want the Medications link in the secondary nav to point to `/my-health/medications/about` while Medications is being deployed, but we still want the secondary nav to highlight the Medications item when the URL is within the app's root URL such as `/my-health/medications`. This change also allows the secondary nav link to be completely outside of an app's root URL, such as when we use the temporary medical records page of `/my-health/records` while the Medical Records app root URL is `/my-health/medical-records`. 

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/83318

## Testing done
- Updated unit tests

### Manual testing
1. Run the landing page locally by running `yarn watch --env --entry=mhv-landing-page`
2. Change the local mocks on the last line of the file `src/applications/mhv-landing-page/mocks/api/index.js` to use `MOCK_TYPES.VERIFIED_USER_ALL_FEATURES`
3. Run the local mock by running `yarn mock-api --responses src/applications/mhv-landing-page/mocks/api/index.js`
4. Access the landing page at http://localhost:3001/my-health/
5. Click on the Medications link and verify it takes you to the Medications about page and that the secondary nav Medications item is highlighted.
6. Navigate to another Medications page like http://localhost:3001/my-health/medications and verify that the secondary nav Medications item is highlighted.

## Screenshots
No changes to the UI made.

## What areas of the site does it impact?
- MHV secondary nav shared component

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback
